bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/internal/scheduler2/CronAdjuster.java

```diff
@@ -0,0 +1,551 @@
+            }
+
+            Temporal out = t.plus(1, type.getBaseUnit());
+
+            switch (type) {
+                case YEAR:
+                    out = out.with(ChronoField.MONTH_OF_YEAR, 1);
+
+                case MONTH_OF_YEAR:
+                    out = out.with(ChronoField.DAY_OF_MONTH, 1);
+
+                case DAY_OF_WEEK:
+                case DAY_OF_MONTH:
+                    out = out.with(ChronoField.HOUR_OF_DAY, 0);
+
+                case HOUR_OF_DAY:
+                    out = out.with(ChronoField.MINUTE_OF_HOUR, 0);
+
+                case MINUTE_OF_HOUR:
+                    out = out.with(ChronoField.SECOND_OF_MINUTE, 0);
+
+                case SECOND_OF_MINUTE:
+                    return out;
+
+                default:
+                    throw new IllegalArgumentException("Invalid field type " + type);
+            }
+
+        }
+    }
+
+    /*
+     * We adjust the given temporal with this type before we check it to prevent
+     * it from keep matching the same value.
+     */
+    TemporalField minIncrement = ChronoField.SECOND_OF_MINUTE;
+
+    final Field seconds;
+    final Field minutes;
+    final Field hours;
+    final Field dayOfMonth;
+    final Field month;
+    final Field dayOfWeek;
+    final Field year;
+    final Field fields[];
+    final Map<String, String> map;
+    final boolean reboot;
+
+    /*
+     * Constructor
+     */
+    public CronAdjuster(String specification) {
+
+        String entries[] = specification.split("(\n|\r)+");
+        map = doEnv(entries);
+
+        String expression = entries[entries.length - 1].trim();
+
+        reboot = expression.equals("@reboot");
+
+        if (expression.startsWith("@")) {
+            expression = preDeclared(expression);
+        }
+
+        String parts[] = expression.trim().toUpperCase().split("\\s+");
+
+        if (parts.length < 6 || parts.length > 7) {
+            throw new IllegalArgumentException(
+                    "Invalid cron expression, too many fields. Only 6 or 7 (with year) allowed: " + expression);
+        }
+
+        seconds = parse(parts[0], ChronoField.SECOND_OF_MINUTE);
+        minutes = parse(parts[1], ChronoField.MINUTE_OF_HOUR);
+        hours = parse(parts[2], ChronoField.HOUR_OF_DAY);
+        dayOfMonth = parse(parts[3], ChronoField.DAY_OF_MONTH);
+        month = parse(parts[4], ChronoField.MONTH_OF_YEAR, "JAN", "FEB", "MAR", "APR", "MAY", "JUN", "JUL", "AUG",
+                "SEP", "OCT", "NOV", "DEC");
+        dayOfWeek = parse(parts[5], ChronoField.DAY_OF_WEEK, "MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN");
+        if (parts.length > 6) {
+            year = parse(parts[6], ChronoField.YEAR);
+        } else {
+            year = null;
+        }
+
+        fields = new Field[] { year, month, dayOfWeek, dayOfMonth, hours, minutes, seconds };
+    }
+
+    private Map<String, String> doEnv(String[] entries) {
+        Map<String, String> map = new HashMap<String, String>();
```

Comment:
Please use the diamond operator, ie replace the second occurrence of the same generic by `<>`